src/dymaptic.GeoBlazor.Core/Components/Layers/ImageryLayer.cs

```diff
@@ -0,0 +1,143 @@
+    ///     Constructor for use in code
+    /// </summary>
+    /// <param name="url"></param>  
+    /// <param name="portalItem"></param>
+    /// <param name="renderer"></param>
+    /// <param name="format"></param>
+    public ImageryLayer(string? url = null, PortalItem? portalItem = null, IImageryRenderer? renderer = null
+        ) 
+    {
+        if (url is null && portalItem is null)
+        {
+            throw new MissingRequiredOptionsChildElementException(nameof(ImageryLayer),
+                new[] { nameof(Url), nameof(PortalItem) });
+        }
+        Renderer = renderer;
+        Url = url;
+        PortalItem = portalItem;
+
+    }
+
+    /// <summary>
+    ///     The url for the Imagery Layer source data.
+    /// </summary>
+    [Parameter]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RequiredProperty(nameof(PortalItem))]
+    public string? Url { get; set; }
+
+    /// <summary>
+    ///     The portal item for the Imagery Layer source data.
+    /// </summary>
+    [RequiredProperty(nameof(Url))]
+    public PortalItem? PortalItem { get; set; }
+
+    /// <summary>
+    ///     An interface that implements the various imagery renderers.
+    /// </summary>
+    public IImageryRenderer? Renderer { get; set; }
+
+    /// <inheritdoc />
+    public override async Task RegisterChildComponent(MapComponent child)
+    {
+        switch (child)
+        {
+            case PortalItem portalItem:
+                if (!portalItem.Equals(PortalItem))
+                {
+                    PortalItem = portalItem;
+                    LayerChanged = true;
+                }
+
+                break;
+            //case IImageryRenderer renderer:
+            //    if (!renderer.Equals(Renderer))
+            //    {
+            //        Renderer = renderer;
+            //        LayerChanged = true;
+            //    }
+            //    break;
+            default:
+                await base.RegisterChildComponent(child);
+
+                break;
+        }
+    }
+    /// <inheritdoc />
+    public override async Task UnregisterChildComponent(MapComponent child)
+    {
+        switch (child)
+        {
+            case PortalItem _:
+                PortalItem = null;
+                LayerChanged = true;
+                break;
+            //case IImageryRenderer _:
+            //    Renderer = null;
+            //    LayerChanged = true;
+            //    break;
+            default:
+                await base.UnregisterChildComponent(child);
+
+                break;
+        }
+    }
+    /// <inheritdoc />
+    internal override void ValidateRequiredChildren()
+    {
+        base.ValidateRequiredChildren();
+        PortalItem?.ValidateRequiredChildren();
+        //Renderer?.ValidateRequiredChildren();
+    }
+}
+
+/// <summary>
+/// The format of the data sent by the server.
+/// </summary>
+[JsonConverter(typeof(EnumToKebabCaseStringConverter<Format>))]
+public enum Format
```

Comment:
This enum has a few issues
- Doesn't appear to be used in this layer, so if it is used at all, it should be co-located with the using layer/component
- The name is probably too vague and will collide with other types in the future. I would suggest naming it something like `ImageFormat` for clarity. You can still name a property as `Format` with type `ImageFormat` for serialization purposes.

Reply:
adjusted